### PR TITLE
[WIP] tests: migrate test 59

### PR DESCRIPTION
**Pull request by lucc**
As mentioned in #3074.

### CODE REVIEW

**Comment by bfredl** [comment]
Awesome work so far :+1: I'll try to look into the remaining errors, but this one of the corners of vim I am least acquainted with...

**Comment by lucc** [comment]
I'm also not acquainted with the spell feature.

Additionally it seems to me that there is some weired stuff going on when passing latin1 and utf8 text between the test suite and nvim. Do you know anything about this? I will try to prepare a test for it.

**Comment by bfredl** [comment]
From my limited understanding, only the `aff/dic` files are latin-1, and the rest of the test is utf-8. All the communication between nvim and lua should be utf-8.

**Comment by lucc** [comment]
But there is some wired conversion going on if I retrieve latin1 text from nvim. See the last test in [this gist](https://gist.github.com/lucc/0393a00957dcaa29f434).

**Comment by bfredl** [comment]
some of those can't possibly work. 

    execute('set fileencoding=latin1')
    feed('i'..latin1..'<ESC>')
    mime('iso-8859-1')
    diff(latin1..'\n')
    expect(latin1)

`fenc=latin1` only mean that nvim will convert internal `utf-8` to `latin-1` , so it will interpret the passed latin-1 as utf-8 and then convert it to latin-1, corrupting the text. As I said, communication between lua and vim must be utf-8 (you can get around it right now by fiddling with `encoding`, like oldtest 59 does, but this won't work after #2929 )

**Comment by bfredl** [comment]
I think you accidentially some utf-8 text, for instance `soundfold('kÃ³opÃ«rÃ¿nÃ´ven')` should be `kóopërÿnôven` . (Not that I blame you, one needs to open `test59.in` with both `++enc=latin-1` and `++enc=utf-8` to copy out the different parts...). I will try to fix it up.

**Comment by bfredl** [comment]
I made some fixes in [bfredl/test59work](https://github.com/bfredl/neovim/tree/test59work), but still some mysterious failures, like
```
bad
-['put', 'OK', 'uk']
+['put', 'uk', 'OK']
 inputs
 ['input', 'puts', 'outputs']
 comment
@@ -32,9 +32,9 @@
 déôl
 ['deol', 'déôr', 'test']
 end
-['put', 'OK', 'uk']
+['put', 'uk', 'test']
 the
-['put', 'OK', 'uk']
+['put', 'uk', 'test']
```
in test 1-1

**Comment by bfredl** [comment]
weird, if I do the setup and `spellsuggest("bad", 3)` etc manually I get test 1-1 right, but not in the lua test... I'll see if same on travis.

**Comment by fmoralesc** [comment]
Hi, @lucc. The original test assumes `syntax off`, it would be nice in regards to #2675 if you could make this assumption explicit. Same goes for tests 45 and 58, if you are working on those.

**Comment by bfredl** [comment]
I can reproduce when running manually, `spellsuggest("bad", 3)` gives different results before and after `:spelldump` which must be a bug. (But why doesn't this happen in the oldtest?)

**Comment by bfredl** [comment]
same bug in vim (as of 7.4.712 )

**Comment by bfredl** [comment]
Aha, just that letting `speldump` be visible on screen affects the scores, as only half of the contents will be shown, and biasing the common word count. in bfredl/test59work I put in a function again, it helps locally at least.

**Comment by bfredl** [comment]
Seems to work on travis as well ( #3103 ), except for unrelated gcov failure...

File: test/functional/legacy/059_utf8_spell_checking_spec.lua [hunk 1]
```diff
@@ -0,0 +1,950 @@
...
+      %yank
+      quit
+      $put
+      $put ='-------'
+      "  Find all bad words and suggestions for them.
+      1;/^]]..aff..[[good:
+      normal 0f:]s
+      let prevbad = ''
+      while 1
+	let [bad, a] = spellbadword()
```
Comment (bfredl) [inline test/functional/legacy/059_utf8_spell_checking_spec.lua]: Shouldn't mix tabs and spaces

**Comment by bfredl** [comment]
Ping @lucc ( If you don't plan to work on this in near time, I'd be happy to take it over)

**Comment by lucc** [comment]
@bfredl yes please do, I will be away for some time soon.

**Comment by bfredl** [comment]
Seems not too much left to do, I will push PR later today or tomorrow.

**Comment by bfredl** [comment]
Continued in #3207

**Pull request closed without merging**